### PR TITLE
Fix cpp scanner regex logic to treat ifndef. This is a PY3.5+ only issue.

### PR DESCRIPTION
**Pull request by ArdaFu**
Previously it was not properly differentiating between if, ifdef, and ifndef

[following last PR 3237](https://github.com/SCons/scons/pull/3237)

## Contributor Checklist:

* [x] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [x] I have updated `master/src/CHANGES.txt` directory (and read the `README.txt` in that directory)
* [x] I have updated the appropriate documentation

### CODE REVIEW

**Comment by bdbaddog** [comment]
Why did you close the last PR and make a new one? now it's hard to follow the discussion...

**Comment by bdbaddog** [comment]
Your changes to the test  must not cover the fix.
If I copy it to plain master branch. it passes.

**Comment by bdbaddog** [comment]
Ahh got it. This is a py3 only issue.

File: src/CHANGES.txt [hunk 1]
```diff
@@ -185,6 +185,9 @@ RELEASE 3.1.0.alpha.yyyymmdd - NEW DATE WILL BE INSERTED HERE
     - Typo in customized decider example in user guide
     - Replace usage of unittest.TestSuite with unittest.main() (fix #3113)
 
+  From Arda Fu
+    - Fix cpp scanner regex logic to treat ifndef. Previously it was not properly
```
Comment (bdbaddog) [inline src/CHANGES.txt]: Can you add that this issue only shows with python 3.5+?

Comment (ArdaFu) [reply src/CHANGES.txt]: You mean like this?

> - Fix cpp scanner regex logic to treat ifndef for py3.5+. Previously 
      it was not properly differentiating between if, ifdef, and ifndef.

**Comment by ArdaFu** [comment]
> Why did you close the last PR and make a new one? now it's hard to follow the discussion...

Sorry but I failed to rebase the official\master. Then, I reset my local repo, modified the code and force pushed to remote end. This removed my last push request commits, and closed the PR automatically. ...

**Comment by bdbaddog** [comment]
Weird. Never had an issue with: git fetch upstream; git rebase upstream/master; get push -f

Seems like regex in python must've changed between py2.7 and 3.5+ to cause this issue?

**Comment by ArdaFu** [comment]
Maybe, but for regex, It looks like an embedded script. Parse regex string should be same for py2.7 and py3.5+.
